File: santa_gives_gifts.cpp
#include<iostream>
using namespace std;
int main()
{
    long long repeat;
    long long AP[5];
    cin>>repeat;
    while(repeat--)
    {
        long long start,stop,a,d,count=0,False=0;
        cin>>start>>stop>>a>>d;
        for(long long term=0;term<5;term++)
            AP[term]=a+(term*d);
        while(start<=stop)
        {
            if(start%AP[0]!=0 && start%AP[1]!=0 && start%AP[2]!=0 && start%AP[3]!=0 && start%AP[4]!=0)
                count++;
            start++;
        }
        cout<<count<<endl;
    }
    return 0;
}

File: santa_gives_gifts.py
repeat=int(input())
while repeat>0:
    ap=list()
    numbers=list()
    start,stop,a,d=map(int,input().split())
    for term in range(5):
        ap.append(a+term*d)
    for num in range(start,stop+1):
        if num%ap[0]!=0:
            numbers.append(num)
    for term in range(1,5):
        for num in numbers:
            if num%ap[term]==0:
                numbers.remove(num)
    print(len(numbers))
    repeat-=1
